NextGenConnect/JSON to HL7v2.xml

```diff
@@ -116,34 +106,320 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.3&apos;] = &quot;&quot;; //not in XML
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.1&apos;] = patient[&apos;name&apos;][0][&apos;family&apos;];
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][0];
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.7&apos;] = &quot;L&quot;;  // Legal Name		
+
+//PID-7 Birthdate 
+
+var birthDate = patient[&apos;birthDate&apos;];
+tmp[&apos;PID&apos;][&apos;PID.7&apos;][&apos;PID.7.1&apos;] = birthDate.split(&apos;-&apos;).join(&apos;&apos;);
+
+
+//PID-8 Administrative Sex -- Needs to be edited for actualy values F/M
+
+if(patient[&apos;gender&apos;] == &apos;female&apos;){  
+	tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = &apos;F&apos;;
+}
+else if(patient[&apos;gender&apos;] == &quot;male&quot;){
+	tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = &apos;M&apos;;
+	}
+
+else{
+	tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = &apos;U&apos;;
+}
+
+//PID-10 Race  -- MVS-528
+
 
-tmp[&apos;PID&apos;][&apos;PID.7&apos;][&apos;PID.7.1&apos;] = patient[&apos;birthDate&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = patient[&apos;gender&apos;][&apos;@value&apos;].toString();
 
 //PID-11 address
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.1&apos;] = patient[&apos;address&apos;][&apos;line&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.7&apos;] = patient[&apos;address&apos;][&apos;use&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.3&apos;] = patient[&apos;address&apos;][&apos;city&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.4&apos;] = patient[&apos;address&apos;][&apos;state&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.5&apos;] = patient[&apos;address&apos;][&apos;postalCode&apos;][&apos;@value&apos;].toString();
 
-//PID-13 Home Phone Number
-tmp[&apos;PID&apos;][&apos;PID.13&apos;][&apos;PID.13.1&apos;] = patient[&apos;telecom&apos;][0][&apos;value&apos;][&apos;@value&apos;].toString();
+var lines = [];
+for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;address&apos;][0][&apos;line&apos;]); i++) {
+	var line = [];
+	line[i] = patient[&apos;address&apos;][0][&apos;line&apos;][i];
+	lines.push(line.join(&apos;, &apos;));
+}
+
 
-//PID-14 Work Phone Number
-tmp[&apos;PID&apos;][&apos;PID.14&apos;][&apos;PID.14.1&apos;] = patient[&apos;telecom&apos;][1][&apos;value&apos;][&apos;@value&apos;].toString();
-*/</script>
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.1&apos;] = lines.join(&apos;&apos;);
```

Comment:
Can this just be like this? It would be simpler than having a for loop. If this won't work, let me know.

`tmp['PID']['PID.11']['PID.11.1'] = patient['address'][0]['line'].join(', ');`